### PR TITLE
DNS ML enablement

### PR DESCRIPTION
**Pull request by rabarona**
This branch enables DNS for ML pipeline.
Main features are:
- Reading from avro parquet.
- Implements path separation

### CODE REVIEW

File: flow_post_lda.scala [hunk 1]
```diff
@@ -1,270 +1,543 @@
...
+        for (i <- 0 to index) {
+          startValue += partSums(i)
+        }
+        partition.scanLeft((0.0, startValue))((prev, curr) => (curr._1, prev._2 + curr._2)).drop(1)
+      }
+    }
+    sumsRdd.map( elem => (elem._1, elem._2 / numValues))
+}
+
+def distributed_quantiles(quantiles: Array[Double], ecdf: org.apache.spark.rdd.RDD[(Double, Double)]): Array[Double] ={
```
Comment (NathanSegerlind) [inline flow_post_lda.scala]: spark 1.6 has built-in functionality like this,  
in future we should look to moving to their implementations (put debug, maintenance and performance on them)

https://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.ml.feature.QuantileDiscretizer

(it is experimental in 1.6.1 so take w/ grain of salt)

File: flow_post_lda.scala [hunk 2]
```diff
@@ -1,270 +1,543 @@
...
-
-
-
-
-
-
-
-
-
+<<<<<<< HEAD
```
Comment (NathanSegerlind) [inline flow_post_lda.scala]: !!!!

unmerged file

!!! 

there's no way this runs... have you committed and pushed all your changes?

Actually, why are you touching flow_post_lda.scala at all? You should probably just take the 1.0.1 version and keep that...

Comment (rabarona) [inline flow_post_lda.scala]: I'm just looking at this. It happened merging 1.0.1 into intelml. Need to re-do this merge sorry. 
I'm going to take back last commit and try merge again. 
So sorry.

**Comment by rabarona** [comment]
@NathanSegerlind: I already fixed the unmerged file.

File: flow_post_lda.scala (hunk 1 above)
Comment (rabarona) [inline flow_post_lda.scala]: I agree however there won't be support for 1.3.0 anymore, do we need to revisit spark versions we support?

**Pull request merged**